src/frost.rs

```diff
@@ -99,6 +177,8 @@ pub struct GroupCommitment(jubjub::ExtendedPoint);
 ///
 /// To derive a FROST keypair, the receiver of the [`SharePackage`] *must* call
 /// .into(), which under the hood also performs validation.
+#[derive(PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
```

Comment:
I would think about whether having an 'external' data type and an 'internal' data type will make maintaining this implementation easier in the future. I believe this is what Zebra does- @dconnolly @teor2345 can give good guidance on this.

Reply:
Zebra has three layers of types in its messages:
- external message protocol based on Bitcoin `zebra_network::protocol::external`
- internal translation of those messages into a request-response protocol `zebra_network::protocol::internal`
- Zcash types based on the chain spec `zebra_chain`

The external and internal messages are wrappers for the chain types.

Since FROST isn't inheriting a badly-specified legacy network protocol, you could have a much simpler design:
* message header struct
* message types enum containing internal types

And then the serialization layer can deal with:
* applying and removing headers
* any logic around version compatibility

```rust
struct Header {
    version: u16
    ..
}

enum Message {
    Share {
        package: SharePackage
    }
    Sign {
        ..
    }
    ..    
}
```